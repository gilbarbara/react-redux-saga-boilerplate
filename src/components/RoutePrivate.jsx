import React from 'react';
import PropTypes from 'prop-types';
import { Route, Redirect } from 'react-router-dom';

const RoutePrivate = ({ component: Component, isAuthenticated, to, ...rest }) => (
  <Route
    {...rest}
    render={props =>
      isAuthenticated ? (
        <Component {...props} />
      ) : (
        <Redirect
          to={{
            pathname: to,
            state: { redirect: props.location.pathname, isAuthenticated },
          }}
        />
      )
    }
  />
);

RoutePrivate.propTypes = {
  component: PropTypes.oneOfType([PropTypes.func, PropTypes.object]).isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  location: PropTypes.object,
  to: PropTypes.string,
};

RoutePrivate.defaultProps = {
  to: '/',
};

export default RoutePrivate;
