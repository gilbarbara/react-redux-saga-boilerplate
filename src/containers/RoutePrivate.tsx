import React from 'react';
import { Redirect, Route } from 'react-router-dom';

import { RouteProps } from 'types';

function RoutePrivate({ component: Component, isAuthenticated, to = '/', ...rest }: RouteProps) {
  return (
    <Route
      {...rest}
      render={props =>
        isAuthenticated ? (
          <Component {...props} />
        ) : (
          <Redirect
            to={{
              pathname: to,
              state: { redirect: props.location.pathname, isAuthenticated },
            }}
          />
        )
      }
    />
  );
}

export default RoutePrivate;
