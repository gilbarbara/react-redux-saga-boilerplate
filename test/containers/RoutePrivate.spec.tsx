import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';

import RoutePrivate from 'containers/RoutePrivate';

import { render } from 'test-utils';

describe('RoutePrivate', () => {
  it('should redirect for unauthenticated access', () => {
    const { container } = render(
      <Router initialEntries={['/private']}>
        <RoutePrivate
          exact
          path="/private"
          component={() => <div>PRIVATE</div>}
          isAuthenticated={false}
        />
      </Router>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should allow navigation for authenticated access', () => {
    const { container } = render(
      <Router initialEntries={['/private']}>
        <RoutePrivate
          exact
          path="/private"
          component={() => <div>PRIVATE</div>}
          isAuthenticated={true}
        />
      </Router>,
    );

    expect(container).toMatchSnapshot();
  });
});
