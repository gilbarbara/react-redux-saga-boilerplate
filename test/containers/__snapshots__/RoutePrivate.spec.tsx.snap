// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoutePrivate should allow navigation for authenticated access 1`] = `
<div>
  <div>
    PRIVATE
  </div>
</div>
`;

exports[`RoutePrivate should redirect for unauthenticated access 1`] = `
<div>
  <div>
    REDIRECT
  </div>
</div>
`;
